--- libpsurface-1.3.1/configure.ac	2012-09-09 12:52:33.000000000 +0200
+++ libpsurface-1.3.1-new/configure.ac	2013-01-21 13:47:36.878118307 +0100
@@ -3,7 +3,7 @@
 AC_CONFIG_SRCDIR([src])
 AC_CONFIG_AUX_DIR([build-aux])
 AM_INIT_AUTOMAKE([-Wall -Werror foreign silent-rules])
-AM_PROG_AR   # needed for some non-standard archivers
+m4_ifdef([AM_PROG_AR],[AM_PROG_AR])   # needed for some non-standard archivers
 LT_INIT
 
 AC_PROG_CXX
